internal/kafka/internal/workers/clusters_mgr.go

```diff
@@ -595,7 +595,7 @@ func (c *ClusterManager) reconcileKasFleetshardOperator(cluster api.Cluster) err
 	} else {
 		if cluster.ClientID == "" {
```

Comment:
```suggestion
		if cluster.ClientID == "" || cluster.ClientSecret == "" {
```

I think this should be checking the two as we need to update both.